### DISCUSSION TITLE
Lesson 10: Defi & Aave (contract object)

### DISCUSSION THREAD

**Discussion by SJ50** [smartcontractkit/full-blockchain-solidity-course-py#1577] [Q&A]
in get_weth function, we use `weth = interface.IWeth(config["networks"][network.show_active()]["weth_token"])` 
what would be the drawback for using `weth = Contract.from_explorer(config["networks"][network.show_active()]["weth_token"])` ?

when I use `Contract.from_explorer`, I get following warning is that a concern and how can eliminate it?
```
Fetching source of 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2 from api.etherscan.io...
/Users/my_user/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/brownie/network/contract.py:1135: 
BrownieCompilerWarning: 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2: 
target compiler 'v0.4.19+commit.c4cbbb05' cannot be installed or is not supported by Brownie. 
some debugging functionality will not be available.
  warnings.warn(
```

If I want to use `weth = Contract.from_abi(name, weth_ecr20_address, abi)`, what would be best method to load abi.json?

I am using following for `Contract.from_abi`
in `helpful_scripts.py`
```
import json
def get_abi(path):
    with open(path) as abi_file:
        abi = json.load(abi_file)
        return abi
```

in `get_weth.py`
```
abi = get_abi("./interfaces/weth_abi.json")
weth = Contract.from_abi("weth", weth_ecr20_address, abi)
```

**Comment by cromewar** [comment]
Hello @SJ50 what a good question, well I would say that is just the remote compiler version is not compatible with the brownie accepted compilers, however this question got me out of the blue @PatrickAlphaC any thoughts on this?

**Reply by SJ50** [reply]
I have forgot to mention; despite that warning, transactions goes through.
